src/client/cli/cmd/info.cpp

```diff
@@ -26,18 +26,17 @@ namespace cmd = multipass::cmd;
 
 namespace
 {
+// TODO@snapshots move this to common_cli once required by other commands
 std::vector<mp::InstanceSnapshotPair> add_instance_and_snapshot_names(const mp::ArgParser* parser)
 {
-    std::vector<mp::InstanceSnapshotPair> instance_snapshot_names;
+    std::vector<mp::InstanceSnapshotPair> instance_snapshot_names(parser->positionalArguments().count());
```

Comment:
In this case, the difference is irrelevant, so I am not going to ask you to change it any further, but this approach has the downside that you perform a bunch of unnecessary initializations: N pairs and 2N strings. It doesn't work with types that are not default constructible either. Finally, if for any reason you overestimate N, you end up with the wrong size. The canonical approach is:

```
std::vector<T> vec{};
vec.reserve(N);
```

Note that `reserve` this does not change size, only capacity. That's what we need to avoid growing the vector and copying things around. It's a pity that you need 2 lines, but that's the interface of `std::vector`.

Reply:
Right..., that function slipped my mind.